packages/chevrotain/src/scan/lexer_public.ts

```diff
@@ -690,7 +691,11 @@ export class Lexer {
         const errorLine = line
         const errorColumn = column
         let foundResyncPoint = false
-        while (!foundResyncPoint && offset < orgLength) {
+        while (
+          this.config.recoveryEnabled &&
+          !foundResyncPoint &&
+          offset < orgLength
+        ) {
```

Comment:
If `recoveryEnabled` is `false`, this will not advance the offset, which then leads to another matching error, which will not advance the offset, leading into... We need to actually stop the lexing process instead of just preventing error recovery.